client/src/MyBookingsPage/SearchBookingComponent.tsx

```diff
@@ -1,66 +1,53 @@
-import { useState, useEffect } from "react";
+import { useState } from "react";
 import Box from "@mui/material/Box";
 import TextField from "@mui/material/TextField";
 
 interface Props {
-  searchFuntion: (email: string, bookingId: string) => void;
+  searchFunction: (email: string, bookingId: string) => void;
+  searchErrorFunction: () => void;
 }
 
-const SearchBookingComponent = ({ searchFuntion }: Props) => {
+const SearchBookingComponent = ({
+  searchFunction,
+  searchErrorFunction,
+}: Props) => {
   const [emailError, setEmailError] = useState(false);
   const [bookingIdError, setBookingIdError] = useState(false);
   const [email, setEmail] = useState("");
   const [bookingId, setBookingId] = useState("");
   const [bookingIdErrorMessage, setBookingIdErrorMessage] = useState("");
   const [emailErrorMessage, setEmailErrorMessage] = useState("");
-  const [searchReady, setSearchReady] = useState(false);
-
-  useEffect(() => {
-    if (!searchReady) return;
-    if (!emailError && !bookingIdError) {
-      searchFuntion(email, bookingId);
-    }
-    setSearchReady(false);
-  }, [
-    searchReady,
-    emailError,
-    bookingIdError,
-    searchFuntion,
-    email,
-    bookingId,
-  ]);
 
   function validateBookingId(id: string) {
-    if (id.length > 1) {
-      setBookingIdError(false);
-      setBookingIdErrorMessage("");
-    } else {
-      setBookingIdError(true);
-      setBookingIdErrorMessage("Incorrect entry. ");
-    }
+    return id.length > 1;
   }
   function validateEmail(emailToValidate: string) {
-    const emailRegex = /\S+@\S+/;
-    if (String(emailToValidate).toLowerCase().match(emailRegex)) {
-      setEmailError(false);
-      setEmailErrorMessage("");
-    } else {
-      setEmailError(true);
-      setEmailErrorMessage("Incorrect entry. ");
-    }
+    const validationPattern = /^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/;
+    return validationPattern.test(emailToValidate);
   }
+
   function search() {
-    validateBookingId(bookingId);
-    validateEmail(email);
-    setSearchReady(true);
+    let idPassedValidation = validateBookingId(bookingId);
+    let emailPassedValidation = validateEmail(email);
+    setBookingIdError(!idPassedValidation);
+    setBookingIdErrorMessage(
+      idPassedValidation ? "" : "Ogiltigt bokningsnummer. "
+    );
+    setEmailError(!emailPassedValidation);
+    setEmailErrorMessage(emailPassedValidation ? "" : "Ogiltigt email. ");
```

Comment:
To be really picky 'Ogiltlig email.' without the ending t :)

Reply:
Grammar has been fixed.